native/src/main/java/io/ballerina/stdlib/jsonpath/BJsonProvider.java

```diff
@@ -0,0 +1,271 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+ package io.ballerina.xlibb.jsonpath;
+
+ import com.jayway.jsonpath.InvalidJsonException;
+ import com.jayway.jsonpath.JsonPathException;
+ import com.jayway.jsonpath.spi.json.AbstractJsonProvider;
+ import io.ballerina.runtime.api.creators.ValueCreator;
+ import io.ballerina.runtime.api.PredefinedTypes;
+ import io.ballerina.runtime.api.types.Type;
+ import io.ballerina.runtime.api.utils.JsonUtils;
+ import io.ballerina.runtime.api.utils.StringUtils;
+ import io.ballerina.runtime.api.values.*;
+ 
+ import java.io.InputStream;
+ import java.math.BigDecimal;
+ import java.util.ArrayList;
+ import java.util.Collection;
+ import java.util.List;
+ 
+ /**
+  * Provides Ballerina specific function implementations of json-path.
+  */
+ public class BJsonProvider extends AbstractJsonProvider {
+     /**
+      * Parse the given json string
+      *
+      * @param json json string to parse
+      * @return Object representation of json
+      * @throws InvalidJsonException
+      */
+     @Override
+     public Object parse(String json) throws InvalidJsonException {
+         return JsonUtils.parse(json);
+     }
+ 
+     /**
+      * Parse the given json bytes in UTF-8 encoding
+      *
+      * @param json json bytes to parse
+      * @return Object representation of json
+      * @throws InvalidJsonException
+      */
+     @Override
+     public Object parse(byte[] json) throws InvalidJsonException {
+         return super.parse(json);
+     }
+ 
+     /**
+      * Parse the given json string
+      *
+      * @param jsonStream input stream to parse
+      * @param charset    charset to use
+      * @return Object representation of json
+      * @throws InvalidJsonException
+      */
+     @Override
+     public Object parse(InputStream jsonStream, String charset) throws InvalidJsonException {
+         return JsonUtils.parse(jsonStream, charset);
+     }
+ 
+     /**
+      * Convert given json object to a json string
+      *
+      * @param obj object to transform
+      * @return json representation of object
+      */
+     @Override
+     public String toJson(Object obj) {
+         return StringUtils.getJsonString(obj);
+     }
+ 
+     /**
+      * Creates a provider specific json array
+      *
+      * @return new array
+      */
+     @Override
+     public Object createArray() {
+         return ValueCreator.createArrayValue(PredefinedTypes.TYPE_JSON_ARRAY);
+     }
+ 
+     /**
+      * Creates a provider specific json object
+      *
+      * @return new object
+      */
+     @Override
+     public Object createMap() {
+         return ValueCreator.createMapValue();
+     }
+ 
+     public boolean isArray(Object obj) {
+         return (obj instanceof BArray);
```

Comment:
Do we need the paranthesis
```suggestion
         return obj instanceof BArray;
```